src/manifolds/SymplecticGrassmannStiefel.jl

```diff
@@ -0,0 +1,158 @@
+@doc raw"""
+    check_point(M::SymplecticGrassmann, p; kwargs...)
+
+Check whether `p` is a valid point on the [`SymplecticGrassmann`](@ref),
+``\operatorname{SpGr}(2n, 2k)`` manifold by verifying that it
+is a valid representer of an equivalence class of the corersponding
+[`SymplecticStiefel`](@ref) manifold.
+"""
+function check_point(M::SymplecticGrassmann, p; kwargs...)
+    n, k = get_parameter(M.size)
+    return check_point(SymplecticStiefel(2 * n, 2 * k), p; kwargs...)
+end
+
+@doc raw"""
+    check_vector(M::SymplecticGrassmann, p, X; kwargs...)
+
+Check whether `X` is a valid tangent vector at `p` on the [`SymplecticGrassmann`](@ref),
+``\operatorname{SpGr}(2n, 2k)`` manifold by verifying that it
+is a valid representer of an equivalence class of the corersponding
+[`SymplecticStiefel`](@ref) manifolds tangent space at `p`.
+"""
+function check_vector(M::SymplecticGrassmann, p, X; kwargs...)
+    n, k = get_parameter(M.size)
+    return check_vector(SymplecticStiefel(2 * n, 2 * k), p, X; kwargs...)
+end
+
+embed(::SymplecticGrassmann, p) = p
+embed(::SymplecticGrassmann, p, X) = X
+embed!(::SymplecticGrassmann, q, p) = copyto!(q, p)
+embed!(::SymplecticGrassmann, Y, p, X) = copyto!(Y, X)
+embed!(::SymplecticGrassmann, q, p::StiefelPoint) = copyto!(q, p.value)
+embed!(::SymplecticGrassmann, Y, p::StiefelPoint, X::StiefelTVector) = copyto!(Y, X.value)
+embed(::SymplecticGrassmann, p::StiefelPoint) = p.value
+embed(::SymplecticGrassmann, p::StiefelPoint, X::StiefelTVector) = X.value
+
+@doc raw"""
+    exp(::SymplecticGrassmann, p, X)
+    exp!(M::SymplecticGrassmann, q, p, X)
+
+Compute the exponential mapping
+
+```math
+  \exp\colon T\mathrm{SpGr}(2n, 2k) → \mathrm{SpGr}(2n, 2k)
+```
+
+when representing points and tangent vectors as symplectic bases and their tangents, i.e.
+on the [`SymplecticStiefel`](@ref) manifold. Then we can just pass this on to [`exp(::SymplecticStiefel, p, X)`](@ref).
+"""
+exp(::SymplecticGrassmann, p, X)
+
+function exp!(M::SymplecticGrassmann, q, p, X)
+    n, k = get_parameter(M.size)
+    exp!(SymplecticStiefel(2 * n, 2 * k), q, p, X)
+    return q
+end
+
+function get_embedding(::SymplecticGrassmann{TypeParameter{Tuple{n,k}},𝔽}) where {n,k,𝔽}
+    return SymplecticStiefel(2n, 2k, 𝔽)
+end
+function get_embedding(M::SymplecticGrassmann{Tuple{Int,Int},𝔽}) where {𝔽}
+    n, k = get_parameter(M.size)
+    return SymplecticStiefel(2n, 2k, 𝔽; parameter=:field)
+end
+
+@doc raw"""
+    inner(::SymplecticGrassmann, p, X, Y)
+
+Compute the Riemannian inner product ``g^{\mathrm{SpGr}}_p(X,Y)``, where ``p``
+is a point on the [`SymplecticStiefel`](@ref) manifold and ``X,Y \in \mathrm{Hor}_p^π\operatorname{SpSt}(2n,2k)``
```

Comment:
I think this should be Grassmann, not Stiefel?

Reply:
the `SymplecticStiefel`? Complicated ;) Similar to Stiefel, when you represent SpGr in SpSt-points (think of similar to p on Grassman represented by one of its ONBs) then this correct in that form, but one could add, that p is a SpGr _represented_ in a SpSt. (other variant: represent p by its projector)

Reply:
I see, I think adding the comment about representing in SpSt would make it more clear.

Reply:
I will have to think how to best phrase that without writing half a paper.

Reply:
The new version reads well enough :+1: